Makefile

```
@@ -120,80 +133,87 @@ profile-module:
 	fi
 
 	@# Cleanup
-	@rm -f $(TEMP_DIR)/typecheck_output.txt
-
+	@rm -f ./$(AGDA_PROFILING_OUTPUT)
 
-
-agda-html: ./src/everything.lagda.md
-	@rm -rf ./docs/
-	@mkdir -p ./docs/
+agda-html: ./$(SOURCE_DIR)/everything.lagda.md
+	@rm -rf ./$(MDBOOK_SRC)/
+	@mkdir -p ./$(MDBOOK_SRC)/
 # Use bash if instead of ifeq, because we want to change SKIPAGDA with rules
 	@if [ -z "$(SKIPAGDA)" ];\
-	then ${AGDA} ${AGDAHTMLFLAGS} ./src/everything.lagda.md; \
-	else python3 ./scripts/generate_noagda_html.py ./docs/; \
+	then ${AGDA} ${AGDAHTMLFLAGS} ./$(SOURCE_DIR)/everything.lagda.md; \
+	else python3 ./$(SCRIPTS_DIR)/generate_noagda_html.py ./$(MDBOOK_SRC)/; \
 	fi
 
-SUMMARY.md: ${AGDAFILES} ./scripts/generate_main_index_file.py
-	@python3 ./scripts/generate_main_index_file.py
+./$(MDBOOK_SRC)/SUMMARY.md: ${AGDAFILES} ./$(SCRIPTS_DIR)/generate_mdbook_summary.py
+	@python3 ./$(SCRIPTS_DIR)/generate_mdbook_summary.py $@
 
-MAINTAINERS.md: ${CONTRIBUTORS_FILE} ./scripts/generate_maintainers.py
-	@python3 ./scripts/generate_maintainers.py
+./$(MDBOOK_SRC)/MAINTAINERS.md: ${CONTRIBUTORS_FILE} ./$(SCRIPTS_DIR)/generate_maintainers.py
+	@python3 ./$(SCRIPTS_DIR)/generate_maintainers.py ${CONTRIBUTORS_FILE} $@
 
-CONTRIBUTORS.md: ${AGDAFILES} ${CONTRIBUTORS_FILE} ./scripts/generate_contributors.py
-	@python3 ./scripts/generate_contributors.py
+./$(MDBOOK_SRC)/CONTRIBUTORS.md: ${AGDAFILES} ${CONTRIBUTORS_FILE} ./$(SCRIPTS_DIR)/generate_contributors.py
+	@python3 ./$(SCRIPTS_DIR)/generate_contributors.py ${CONTRIBUTORS_FILE} $@
 
-website/css/Agda-highlight.css: ./scripts/generate_agda_css.py ./theme/catppuccin.css
-	@python3 ./scripts/generate_agda_css.py
+$(WEBSITE_CSS)/Agda-highlight.css: ./$(SCRIPTS_DIR)/generate_agda_css.py ./$(WEBSITE_THEME)/catppuccin.css
+	@python3 ./$(SCRIPTS_DIR)/generate_agda_css.py
 
-website/images/agda_dependency_graph.svg website/images/agda_dependency_graph_legend.html &: ${AGDAFILES}
-	@python3 ./scripts/generate_dependency_graph_rendering.py website/images/agda_dependency_graph svg || true
+$(WEBSITE_IMAGES)/agda_dependency_graph.svg $(WEBSITE_IMAGES)/agda_dependency_graph_legend.html &: ${AGDAFILES}
+	@python3 ./$(SCRIPTS_DIR)/generate_dependency_graph_rendering.py $(WEBSITE_IMAGES)/agda_dependency_graph svg || true
 
 .PHONY: website-prepare
-website-prepare: agda-html ./SUMMARY.md ./CONTRIBUTORS.md ./MAINTAINERS.md \
-								 ./website/css/Agda-highlight.css $(AUX_TARGETS)
-	@cp $(METAFILES) ./docs/
-	@mkdir -p ./docs/website
-	@cp -r ./website/images ./docs/website/
-	@cp -r ./website/css ./docs/website/
-	@cp -r ./website/js ./docs/website/
-	@cp -r ./tables ./docs
+website-prepare: agda-html ./$(MDBOOK_SRC)/SUMMARY.md ./$(MDBOOK_SRC)/CONTRIBUTORS.md ./$(MDBOOK_SRC)/MAINTAINERS.md \
+								 ./$(WEBSITE_CSS)/Agda-highlight.css $(AUX_TARGETS)
+	@rm -f ./docs/CONTRIBUTORS.md ./docs/MAINTAINERS.md ./docs/SUMMARY.md # Remove old autogenerated metafile copies
+
+	@mkdir -p ./$(MDBOOK_SRC)
+	@mkdir -p ./$(MDBOOK_SRC)/website
```

```suggestion
	@mkdir -p ./$(MDBOOK_SRC)/website
```
The first line is subsumed by the second